### PR TITLE
Jonk lockfix

### PR DESCRIPTION
**Pull request by jonkelleyatrackspace**
Has the foolproof lockfile fix

### CODE REVIEW

File: url_monitor/main.py [hunk 1]
```diff
@@ -130,79 +131,83 @@ def main(arguments=None):
...
 
     if inputflag.COMMAND == "check":
-        # establish single-run lockfile (pid)
-        try:
-            runlock = commons.AcquireRunLock(config['config']['pidfile'])
-        except PidlockConflict, err:
-            logging.error("Error: Could not acquire exclusive "
-                          "lock {0}".format(err))
-            print("1")
+        lock = lockfile.FileLock(config['config']['pidfile'])
```
Comment (slick666) [inline url_monitor/main.py]: Are we doing any validation that config had these fields? Do we want to add a clear try block around this?

File: url_monitor/main.py [hunk 2]
```diff
@@ -130,79 +131,83 @@ def main(arguments=None):
...
-            logger.critical(
-                "Sending execution summary to zabbix server failed!")
-            set_rc = 1
+        else:
+            with lock:
+                logger.info("PID lock acquired {0} {1}".format(lock.path, lock.pid))
+
+                # run check
+                completed_runs = [] # check results
+                for thisscheck in config['checks']:
```
Comment (slick666) [inline url_monitor/main.py]: what does "this s check" mean?

File: url_monitor/main.py [hunk 3]
```diff
@@ -130,79 +131,83 @@ def main(arguments=None):
...
+                    try:
+                        if (inputflag.key != None and
```
Comment (slick666) [inline url_monitor/main.py]: use "is not" instead of "!=" when referring to None

File: url_monitor/main.py (hunk 1 above)
Comment (jonkelleyatrackspace) [inline url_monitor/main.py]: Do we even know all the exception cases? I will add a generic handler

Comment (teriyakichild) [inline url_monitor/main.py]: https://github.com/rackerlabs/zabbix_url_monitor/blob/master/url_monitor/configuration.py#L642-L646

File: url_monitor/main.py (hunk 3 above)
Comment (teriyakichild) [inline url_monitor/main.py]: Actually, all of this code already existed.  The change is the indentation.

Comment (jonkelleyatrackspace) [inline url_monitor/main.py]: ok

File: url_monitor/main.py (hunk 1 above)
Comment (slick666) [inline url_monitor/main.py]: It should just be a simple KeyError.
https://docs.python.org/2/library/exceptions.html#exceptions.KeyError
Check the yaml documentation and validate but that should cover either key not existing

File: url_monitor/main.py (hunk 2 above)
Comment (slick666) [inline url_monitor/main.py]: I understand, I was just thinking it was a type-o "thischeck" vs "thisscheck".

Comment (jonkelleyatrackspace) [inline url_monitor/main.py]: good catch i cant read

File: url_monitor/main.py [hunk 4]
```diff
@@ -130,79 +131,118 @@ def main(arguments=None):
...
-            runlock = commons.AcquireRunLock(config['config']['pidfile'])
-        except PidlockConflict, err:
-            logging.error("Error: Could not acquire exclusive "
-                          "lock {0}".format(err))
-            print("1")
+            lock = lockfile.FileLock(config['config']['pidfile'])
+        except lockfile.NotLocked as e:
+            logger.error(
+                "lockfile exception: the lock is not locked when release() was called? {0}".format(e))
+            echo("1")
```
Comment (teriyakichild) [inline url_monitor/main.py]: I think "echo" should be "print"?  If so, why are we printing 1 to stdout?

File: url_monitor/main.py [hunk 5]
```diff
@@ -130,79 +131,118 @@ def main(arguments=None):
...
+            exit(1)
+        except lockfile.NotLocked as e:
+            logger.error(
+                "lockfile exception: the lock is was locked when release() was called {0}".format(e))
+            echo("1")
+            exit(1)
+        except lockfile.NotMyLock as e:
+            logger.error(
+                "lockfile exception: a lock already exists, but appears to be owned by another process! {0}".format(e))
```
Comment (teriyakichild) [inline url_monitor/main.py]: Wouldn't the lockfile always be owned by a different process.  I believe NotMyLock should indicate that the lockfile is owned by a different user.

File: url_monitor/main.py [hunk 6]
```diff
@@ -130,79 +131,118 @@ def main(arguments=None):
...
     if inputflag.COMMAND == "check":
-        # establish single-run lockfile (pid)
         try:
-            runlock = commons.AcquireRunLock(config['config']['pidfile'])
-        except PidlockConflict, err:
-            logging.error("Error: Could not acquire exclusive "
-                          "lock {0}".format(err))
-            print("1")
+            lock = lockfile.FileLock(config['config']['pidfile'])
+        except lockfile.NotLocked as e:
```
Comment (teriyakichild) [inline url_monitor/main.py]: Maybe I don't understand how this works, but I would think that "NotLocked" would be a very common situation.  Why are we throwing an error here?

Comment (jonkelleyatrackspace) [inline url_monitor/main.py]: gone

**Pull request merged**